recirq/fermi_hubbard/publication.py

```diff
@@ -239,23 +240,26 @@ def fetch_publication_data(
     if base_dir is None:
         base_dir = "fermi_hubbard_data"
 
-    base_url = "https://datadryad.org/stash/downloads/file_stream/"
-    data = {
-        "gaussians_1u1d_nofloquet": "451326",
-        "gaussians_1u1d": "451327",
-        "trapping_2u2d": "451328",
-        "trapping_3u3d": "451329"
-    }
-    if exclude is not None:
-        data = {path: key for path, key in data.items() if path not in exclude}
-
-    for path, key in data.items():
+    fnames = {
+        "gaussians_1u1d_nofloquet", "gaussians_1u1d", "trapping_2u2d", "trapping_3u3d"
+    }.difference(exclude)
+
+    # Determine file IDs. Note these are not permanent on Dryad.
+    file_ids = {}
+    for line in urlopen("https://doi.org/10.5061/dryad.crjdfn32v").readlines():
```

Comment:
is this parsing the human-readable html website? If so, this seems just as unstable. What if they add a newline between the filename.zip and the id?

Reply:
Yes this would break it. It might be that this is more stable than the file IDs - the file IDs changed at least once since `fetch_publication_data` was added, but I don't think the html has. I think the only way to hermetically seal it is to tell Dryad to give individual files DOIs, or use a different service that does.